src/main/java/io/phdata/streamliner/schemadefiner/model/TableDiff.java

```diff
@@ -38,8 +39,61 @@ public TableDiff(
     this.existsInSource = existsInSource;
   }
 
-  public boolean allChangesAreCompatible(){
-    return columnDiffs.stream().allMatch(c -> c.getIsAdd());
+  public boolean allChangesAreCompatible(scala.collection.immutable.Map<String, scala.collection.immutable.Map<String, String>>
+                                                 typeMapping) {
+    Map<String, Map<String, String>> javaTypeMap = JavaHelper.convertScalaMapToJavaMap(typeMapping);
+    boolean flag = true;
+    for (ColumnDiff colDiff : columnDiffs) {
+      if (colDiff.getIsDeleted()) {
+        flag = false;
+        break;
+      } else if (colDiff.getIsUpdate()) {
```

Comment:
I think this code would be a little simpler if just returned false and below. Then eliminate the flag variable.